pkgs/development/interpreters/python/hooks/unittest-check-hook.sh

```diff
@@ -0,0 +1,29 @@
+# Setup hook for unittest.
+echo "Sourcing unittest-check-hook"
+
+unittestCheckPhase() {
+    echo "Executing unittestCheckPhase"
+    runHook preCheck
+
+    eval "@pythonCheckInterpreter@ -m unittest discover $unittestFlags"
```

Comment:
I'm not sure if this is the best way to do things.

The issue that I ran into was that if I did `python -m unittest discover "${unittestFlags[@]}"`, a flags value of `[ "-s" "tests" ]` would expand to a single argument, `-s tests`, which `argparse` (which `unittest` uses) would parse as the value for the `-s` argument being ` tests` (with a space).

Reply:
maybe this https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/hooks/pytest-check-hook.sh#L52-L53 helps you?

Reply:
That's basically doing what I'm doing here, as far as I can tell -- I believe it has to do that specifically so it can add its own arguments if needed.